File: DSA-Python/2-patterns/15-pattern.py
'''
ABCDE
ABCD
ABC
AB
A
'''

def main(n:int)-> None:
    for i in range(n, 0, -1):
        for j in range(1, i+1):
            print(chr(64+j), end='')
            
        print()

if __name__ == "__main__": 
    n = int(input())
    main(n)

File: DSA-C++/3-patterns/15-pattern.cpp
#include <bits/stdc++.h>

using namespace std;

// ABCDE
// ABCD
// ABC
// AB
// A

int main() {
    int n;
    cin >> n;

    for (int i=n; i>=1; --i) {
        int c = 64;
        for (int j=i; j>=1; --j) {
            cout << static_cast <char> (++c);
        }

        cout << endl;
    }

    return 0;
}